packages/flutter/lib/src/material/scaffold.dart

```diff
@@ -1299,6 +1317,16 @@ class Scaffold extends StatefulWidget {
   /// 20.0 will be added to `MediaQuery.of(context).padding.left`.
   final double drawerEdgeDragWidth;
 
+  /// This flag is deprecated and fixes and issue with incorrect clipping
+  /// and positioning of the [SnackBar] set to [SnackBarBehavior.floating].
+  @Deprecated(
+    'Fixes a bug that that fixes clipping and positioning of SnackBar '
+    'as it incorrectly offsets itself assuming a floating action button '
+    'is present when it is not. This parameter will be removed. '
+    'This feature was deprecated after v1.15.3.'
+  )
+  final bool shouldSnackBarIgnoreFABRect;
```

Comment:
Nit - why not just make this static?

Reply:
How would that work? Would all Flutter apps then simply set `Scaffold.shouldSnackBarIgnoreFABRect = true` before they call `runApp`?

Reply:
Right.  It's not something we expect people to want to use, let alone to use differently for different scaffolds.

Presumably they could even set it after runApp, but they should probably do it before any builds kick off to avoid weirdness.

Reply:
Done